lib/unnestN/index.js

```diff
@@ -1,13 +1,17 @@
 import curry2 from '../_internal/_curry2'
+import compose from '../compose'
 import unnest from '../unnest'
+import not from '../not'
 
 export default curry2(function unnestN(num, arr) {
-  return Array.isArray(arr)
-    ? num === 0 ||
-      arr.every(function(item) {
-        return !Array.isArray(item)
-      })
-      ? arr
-      : unnestN(num - 1, unnest(arr))
-    : []
+  if (!Array.isArray(arr) || num < 0) {
+    return []
+  }
+  var res = arr
+  var i = 0
+  while (i < num && !res.every(compose(not, Array.isArray))) {
```

Comment:
I think it's much more expensive to check `!res.every(compose(not, Array.isArray))`, than just let it do several more `concat` operations in `unnest` method
It's worth checking

Reply:
also I'm thinking of moving `compose(not, Array.isArray)` away from `every` as it's created over and over again for each iteration (maybe)
at least I need to check it twice

Reply:
Yep, that's the only truthy way

Reply:
> also I'm thinking of moving `compose(not, Array.isArray)` away from `every` as it's created over and over again for each iteration (maybe)
> at least I need to check it twice

Good thought